### PR TITLE
Docs changes to BucketSentenceIter, Optimizer, ImageIter

### PR DESCRIPTION
**Pull request by lxn2**

### CODE REVIEW

File: python/mxnet/image.py [hunk 1]
```diff
@@ -351,8 +351,8 @@ class ImageIter(io.DataIter):
     data_shape : tuple
         Data shape in (channels, height, width) format.
         For now, only RGB image with 3 channels is supported.
-    label_width : int
-        Label dimension.
+    label_width : int, optional
+        Label dimension. The default label width is 1.
```
Comment (piiswrong) [inline python/mxnet/image.py]: Label dimension (number of labels per example)

File: python/mxnet/optimizer.py [hunk 2]
```diff
@@ -154,18 +154,18 @@ def create_state(self, index, weight):
...
 
     def update(self, index, weight, grad, state):
-        """Update the weight given the corresponding gradient and state.
+        """Update the parameters given the corresponding gradients and state.
 
         Parameters
         ----------
         index : int
-            An unique index to identify the weight.
+            A unique index to identify the parameter.
```
Comment (piiswrong) [inline python/mxnet/optimizer.py]: explain relationship to set_lr_mult and set_wd_mult

File: python/mxnet/image.py (hunk 1 above)
Comment (zackchase) [inline python/mxnet/image.py]: Does this mean the number of classes? Perhaps we can be clearer? Right now we say "Label dimension" and "label width" ostensibly to refer to the same quantity. Let's use consistent notation.

File: python/mxnet/image.py [hunk 3]
```diff
@@ -462,6 +462,7 @@ def __init__(self, batch_size, data_shape, label_width=1,
         self.reset()
 
     def reset(self):
+        """Reset the iterator to the beginning of the data."""
```
Comment (zackchase) [inline python/mxnet/image.py]: "Reset" -> "Resets"

File: python/mxnet/image.py [hunk 4]
```diff
@@ -493,6 +494,7 @@ def next_sample(self):
             return header.label, img
 
     def next(self):
+        """Return the next batch of data."""
```
Comment (zackchase) [inline python/mxnet/image.py]: "Return" -> "Returns"

File: python/mxnet/optimizer.py [hunk 5]
```diff
@@ -154,18 +154,18 @@ def create_state(self, index, weight):
         """
 
     def update(self, index, weight, grad, state):
-        """Update the weight given the corresponding gradient and state.
+        """Update the parameters given the corresponding gradients and state.
```
Comment (zackchase) [inline python/mxnet/optimizer.py]: "Update" -> "Updates"

Comment (zackchase) [inline python/mxnet/optimizer.py]: Is this updating a single parameter tensor or multiple? If a single one let's just say: "Updates the given parameter using the corresponding gradient and state."

File: python/mxnet/rnn/io.py [hunk 6]
```diff
@@ -135,6 +135,7 @@ def __init__(self, sentences, batch_size, buckets=None, invalid_label=-1,
         self.reset()
 
     def reset(self):
+        """Reset the iterator to the beginning of the data."""
```
Comment (zackchase) [inline python/mxnet/rnn/io.py]: "Reset" -> "Resets"

File: python/mxnet/rnn/io.py [hunk 7]
```diff
@@ -150,6 +151,7 @@ def reset(self):
             self.ndlabel.append(ndarray.array(label, dtype=self.dtype))
 
     def next(self):
+        """Return the next batch of data."""
```
Comment (zackchase) [inline python/mxnet/rnn/io.py]: "Return" -> "Returns"

**Comment by zackchase** [comment]
Hey can we get these changes made so we can merge in?

File: python/mxnet/image.py (hunk 1 above)
Comment (piiswrong) [inline python/mxnet/image.py]: its not number of classes, which is the range of label. Its the number of labels.

**Comment by lxn2** [comment]
@zackchase @piiswrong updated with suggested changes

**Comment by zackchase** [comment]
@piiswrong can we merge?

**Pull request merged**